CPANEL-47086

cpanelrikus opened the pull request:
…roval.

Case CPANEL-47086: A previous fix to decode name into a str before feeding it into formataddr() needed to apply to a second call to formataddr(), so it has been moved to take effect for both.

Changelog: Mailman subscription approval: Fix AttributeError 'bytes' object
  has no attribute 'encode'.

cpanel-jared reviewed (approved):
~LGTM~

cpanel-jared reviewed (changes requested):
Might have found an issue, there is another call to `formataddr()` that looks suspect.

https://github.com/cpanel/mailman2-python3/pull/21/files#diff-a1f9b2b0bea27b1af3e4be3f16fc2d6bf5f77cdbee30144bba0eb69b23fde646L1027

This is a different subscription code path that I don't think gets used often, but it contains a `formataddr()` call that seems like it would fail in the same way as the others.

Commit:
Additional encoding fix for use with formataddr() in subscription approval.

Case CPANEL-47086: A previous fix to decode name into a str before
feeding it into formataddr() needed to also be applied to 2 other
calls to formataddr().

Changelog: Mailman subscription approval: Fix AttributeError
  'bytes' object has no attribute 'encode'.
